src/Components/Components/src/Routing/Router.cs

```diff
@@ -141,6 +141,11 @@ public async Task SetParametersAsync(ParameterView parameters)
             throw new InvalidOperationException($"The {nameof(Router)} component requires a value for the parameter {nameof(Found)}.");
         }
 
+        if (NotFound != null && NotFoundPage != null)
+        {
+            throw new InvalidOperationException($"The {nameof(Router)} component cannot have both {nameof(NotFound)} and {nameof(NotFoundPage)} parameters set. Use either {nameof(NotFound)} render fragment or {nameof(NotFoundPage)} component type, but not both.");
```

Comment:
Use the logger to log warning instead of throwing. Add a new method to `private static partial class Log` with warning level. In the warning message, underline that we prefer `NotFoundPage` to be used and that `NotFound` will be deprecated.